### PR TITLE
LUN-1813

### PR DESCRIPTION
**Pull request by laurafeier**

### CODE REVIEW

File: cms/tests/admin.py [hunk 1]
```diff
@@ -403,6 +404,101 @@ def test_changelist_tree(self):
         # but not any further down the tree
         self.assertFalse('id="page_4"' in response.content)
 
+    def test_changelist_site_items(self):
+        site_one = Site.objects.get_current()
+        site_two = Site.objects.create(
+            domain='django-cms2.org', name='django-cms2')
+        normal_guy = self._get_guys()[1]
```
Comment (pianas) [inline cms/tests/admin.py:411]: Maybe it's a matter of taste, but 'guy', 'dude', 'pal', ... are not as good as 'user' :)
In this case, maybe 'unprivileged_user' (or 'normal_user') is more suitable.

File: cms/admin/change_list.py [hunk 2]
```diff
@@ -56,6 +57,13 @@ class CMSChangeList(ChangeList):
     def __init__(self, request, *args, **kwargs):
         from cms.utils.plugins import current_site
         self._current_site = current_site(request)
+        self.set_sites(request)
+        if self._current_site:
+            is_current = lambda x: x.id == self._current_site.id
+            is_site_allowed = next(ifilter(is_current, self.sites), None)
+            if not is_site_allowed and self.sites:
+                self._current_site = self.sites[0]
+            request.session['cms_admin_site'] = self._current_site.pk
```
Comment (pianas) [inline cms/admin/change_list.py:67]: after reading the code, I guess that the meaning is 'if the current site has changed, then update the session's cms_admin_site param'. But this is not obvious in the first seconds of code reading, thus a comment would be welcome.

File: cms/tests/admin.py (hunk 1 above)
Comment (laurafeier) [inline cms/tests/admin.py:411]: I'm using the same code style as the rest of the tests in this module.

File: cms/utils/permissions.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def get_current_user():
     """
     return getattr(_thread_locals, 'user', None)
 
-def has_page_add_permission(request):
+def has_page_add_permission(request, on_any_sites=False):
```
Comment (pianas) [inline cms/utils/permissions.py:34]: small correction: 'on_any_site' (without 's')

File: cms/admin/change_list.py [hunk 4]
```diff
@@ -56,6 +57,14 @@ class CMSChangeList(ChangeList):
     def __init__(self, request, *args, **kwargs):
         from cms.utils.plugins import current_site
         self._current_site = current_site(request)
+        self.set_sites(request)
+        if self._current_site:
+            is_current = lambda x: x.id == self._current_site.id
+            is_site_allowed = next(ifilter(is_current, self.sites), None)
```
Comment (zoliszeredi) [inline cms/admin/change_list.py:63]: Even though I disagree, PEP 8 advises against named lambdas.
http://legacy.python.org/dev/peps/pep-0008/#programming-recommendations
For the is_site_allowed you may also use a generator expression:

``` python
is_site_allowed = next((site
                        for site in self.sites 
                        if site.id == self._current_site.id), None)
```

File: cms/utils/permissions.py [hunk 5]
```diff
@@ -89,10 +88,13 @@ def has_page_change_permission(request):
     """
     from cms.utils.plugins import current_site
     opts = Page._meta
+    site = current_site(request) if not on_any_sites else None
     if request.user.is_superuser or (
         request.user.has_perm(opts.app_label + '.' + opts.get_change_permission()) and (
-            has_global_page_permission(request, current_site(request), can_change=True))
-            or has_any_page_change_permissions(request)):
```
Comment (titusnicolae) [inline cms/utils/permissions.py:95]: function calls could be assigned to variables for readability

**Pull request merged**